experimental/ModelBuilder/test/BenchMatMulVectorGPU.cpp

```diff
@@ -135,22 +140,124 @@ static SmallVector<linalg::ProcInfo, 2> getSubgroupIds(
   return procInfo;
 }
 
-void matMul(int m, int n, int k, int tileM, int tileN, int tileK,
-            bool correctness) {
+struct MatMulF32 {
```

Comment:
Why should this be wrapped in a struct? You could just do

```
template <typename T>
static mlir::type getMLIRType(MLIRContext &ctx);
```

and specialize for each type...

Reply:
To add, actually here just a utility method with a switch case seems as good.

Reply:
The main reason I used templated struct is because Type needs to be static as it is used it by the makeInitializedStridedMemRefDescriptor functions and I also use it for to pick the type when doing the CPU calculation. 
I agree it doesn't look great but I couldn't find another solution bigger changes. What do you think?

Reply:
Sure. THis is fine. Its in ModelBuilder which is meant for prototyping anyway. So whatever makes it easy for you